display explanations of modifications made to URDF

lewandowskikacper opened the pull request:
Proposition of a way to present modifications of URDF to user:
Resolves https://github.com/o3de/o3de-extras/issues/512.

Note: this PR includes commits from https://github.com/o3de/o3de-extras/pull/511, and should be merged after it.

![image](https://github.com/o3de/o3de-extras/assets/138626322/1c6fda2c-4c26-4905-b55b-6cdfeadd157b)

michalpelka commented:
Please rebase and then set it to review. It is an important change. @lewandowskikacper

jhanca-robotecai commented on Gems/ROS2/Code/Source/RobotImporter/URDF/UrdfParser.h:
```diff
@@ -75,8 +76,8 @@ namespace ROS2
-            //! Stores the modified URDF tags after parsing, empty if no modification occurred
-            AZStd::vector<AZStd::string> m_modifiedURDFTags;
+            //! Stores describtion of URDF modifications, empty if no modification occurred
+            Utils::UrdfModifications m_UrdfModifications;
```
The same change applies for the rest of usages
```suggestion
            Utils::UrdfModifications m_urdfModifications;
```

jhanca-robotecai reviewed (commented):
The changes are looking good, please continue.

adamdbrw commented on Gems/ROS2/Code/Source/RobotImporter/FixURDF/FixURDF.cpp:
```diff
@@ -41,23 +43,36 @@ namespace ROS2::Utils
+                AZ_Warning(
+                    "URDF", !inertialPresent, "Missing mass tag inside the inertial tag link %s, applying default mass tag.", name.c_str());
                 mass = urdf->document()->allocate_node(node_element, "mass");
                 mass->append_attribute(urdf->document()->allocate_attribute("value", "1."));
```
There could be an estimation made from the mesh interior volume here (general comment, I am aware PR does not change that).

lewandowskikacper commented on Gems/ROS2/Code/Source/RobotImporter/RobotImporterWidget.cpp:
```diff
@@ -159,10 +160,36 @@ namespace ROS2
                 {
                     // This is a URDF only path, and therefore the report text does not mention SDF
```
Moved to `AddModificationWarningsToReportString`

lewandowskikacper commented:
I rebased on top of development again. Please note that without https://github.com/o3de/o3de-extras/pull/534 it will not compile.

I also noticed that "The following joints were renamed to avoid duplication" is printed even if no joint was renamed (but e.g. some inertial was missing). I added a simple check for that.
